src/shared/Core/Credential.cs

```diff
@@ -15,21 +18,57 @@ public interface ICredential
         /// Password.
         /// </summary>
         string Password { get; }
+
+        /// <summary>
+        /// The expiry date of the password. This is Git's password_expiry_utc
+        /// attribute. https://git-scm.com/docs/git-credential#Documentation/git-credential.txt-codepasswordexpiryutccode
+        /// </summary>
+        DateTimeOffset? PasswordExpiry { get => null; }
```

Comment:
This may need to be updated to avoid default interface methods given the current situation with .NET 8 on Windows.